### PR TITLE
StaticConfigTrait::getConfig($key) return type

### PR DESCRIPTION
**Pull request by andrii-pukhalevych**
StaticConfigTrait::getConfig($key) may return int/string/float, not only array

### CODE REVIEW

File: StaticConfigTrait.php [hunk 1]
```diff
@@ -110,7 +110,7 @@ public static function setConfig($key, $config = null)
      * Reads existing configuration.
      *
      * @param string $key The name of the configuration.
-     * @return array|null Array of configuration data.
+     * @return array|mixed|null Configuration data.
```
Comment (dereuromark) [inline StaticConfigTrait.php:113]: I dont think this is true. If you see the other methods it is an array of config coming back, or a string.

**Comment by dereuromark** [comment]
Also, PRs to splits cannot be accepted, the PRs should be done against main repo ( https://github.com/cakephp/cakephp ).

File: StaticConfigTrait.php (hunk 1 above)
Comment (andrii-pukhalevych) [reply StaticConfigTrait.php:113]: But it's trait and can be used in different way... I used it with own class.
So why part of config can not be int or string?

**Comment by andrii-pukhalevych** [comment]
Is it possible to rebase current PR from split to to main repo, or I should create new PR?

**Comment by dereuromark** [comment]
It has to be a new PR.

**Comment by andrii-pukhalevych** [comment]
I used this trait with own class. So why part of config can not be int or string?
Will you accept similar PR to main repo?

**Comment by ADmad** [comment]
@andrii-pukhalevych You can make a PR against main repo. One can store and retrieve value of any type so `mixed` type would be correct.

**Pull request closed without merging**